libs/libxx/uClibc++.defs

```diff
@@ -20,22 +20,29 @@
 
 UCLIBCXX_VERSION=0.2.5
 
-$(TOPDIR)/include/uClibc++:
+# Download and unpack tarball if no git repo found
+ifeq ($(wildcard uClibc++/.git),)
+uClibc++:
 	curl -O -L https://git.busybox.net/uClibc++/snapshot/uClibc++-$(UCLIBCXX_VERSION).tar.bz2
 	$(Q) tar -xf uClibc++-$(UCLIBCXX_VERSION).tar.bz2
 	$(Q) $(DELFILE) uClibc++-$(UCLIBCXX_VERSION).tar.bz2
 	$(Q) mv uClibc++-$(UCLIBCXX_VERSION) uClibc++
-	$(Q) $(DIRLINK) $(CURDIR)/uClibc++/include $(TOPDIR)/include/uClibc++
-	$(Q) $(COPYFILE) $(CURDIR)/system_configuration.h $(TOPDIR)/include/uClibc++
 	$(Q) patch -p0 < 0001-uClibcxx-basic_definitions-fix-GCC-specific-definiti.patch
 	$(Q) patch -p0 < 0001-uclibxx-use-overload-constructor-of-filebuf-ostream.patch
+endif
+
+$(TOPDIR)/include/uClibc++: uClibc++
+	$(Q) $(DIRLINK) $(CURDIR)/uClibc++/include $(TOPDIR)/include/uClibc++
+	$(Q) $(COPYFILE) $(CURDIR)/system_configuration.h $(TOPDIR)/include/uClibc++
 
 context:: $(TOPDIR)/include/uClibc++
 
 distclean::
 	$(Q) $(DELFILE) $(TOPDIR)/include/uClibc++/system_configuration.h
 	$(Q) $(DIRUNLINK) $(TOPDIR)/include/uClibc++
+ifeq ($(wildcard uClibc++/.git),)
```

Comment:
Since now it bypasses distclean, how does it get updated?

Reply:
If you use repo or submodule integrate the 3rd party library into source control system, you need update the 3rd party library through git command(e.g. merge or rebase), and never e txpect that he build system remove your git in distclean.

Reply:
The version kconfig option that we usually add to these external libraries will be a bit misleading I believe.  Maybe we can make `context` do as you described if the version already downloaded is not similar to the one requested.  But this will bloat it and requires some `git` manipulation or whatever the library provides.
(not asking to do it, just thinking out load :) )

Reply:
> The version kconfig option that we usually add to these external libraries will be a bit misleading I believe.

The version is only useful if the user select to download the 3rd party library from internet.

> Maybe we can make `context` do as you described if the version already downloaded is not similar to the one requested.

Yes, the version config doesn't support the incremental build, which should be improved.

> But this will bloat it and requires some `git` manipulation or whatever the library provides. (not asking to do it, just thinking out load :) )

It's obvious that version option doesn't make sense if user select to manage the 3rd party library with git.

Reply:
@Ouss4 do you have more concern?